node/src/types/validator_matrix.rs

```diff
@@ -92,6 +92,27 @@ impl ValidatorMatrix {
         }
     }
 
+    // When the chain starts, the validator weights will be the same until the unbonding delay is
+    // elapsed. This allows us to possibly infer the weights of other eras if the era registered is
+    // within the unbonding delay.
+    // Currently we only infer the validator weights for era 0 from the set registered for era 1.
+    // This is needed for the case where we want to sync leap to a block in era 0 of a pre 1.5.0
+    // network for which we cant get the validator weights from a switch block.
+    pub(crate) fn register_era_validator_weights_and_infer_era_0(
+        &mut self,
+        validators: EraValidatorWeights,
+    ) -> bool {
+        let was_present = self.register_era_validator_weights(validators.clone());
+        if validators.era_id() == EraId::from(1) {
+            self.register_era_validator_weights(EraValidatorWeights::new(
```

Comment:
Can you add an "info" or "debug" level log here, saying "infering validator weights for era 0"?

Reply:
Info level seems a bit excessive I think but a debug log is a good ideea.

Reply:
added in https://github.com/casper-network/casper-node/pull/3639/commits/7a7a7adc38ff918e87fc1265f7bef868149c1b54